File: src/BinaryDataDecoders.CodeAnalysis.Tests/CSharp/SampleClasses.cs
using System;
using System.Collections.Generic;
using System.Runtime.InteropServices;
using System.Text;

namespace BinaryDataDecoders.CodeAnalysis.Tests.CSharp
{
    //line comment
    [Serializable]
    class SampleClass_NoVisibility
    {
        /// <summary>
        /// xml comment
        /// </summary>
        [Obsolete]
        public SampleClass_NoVisibility()
        {
        }
        /* block
         * comment */
        public SampleClass_NoVisibility([In] int value)
        {
        }

        static SampleClass_NoVisibility()
        {
        }

        ~SampleClass_NoVisibility()
        {
        }

        public string PropertyGetterOnlyArrow => null;
        public string PropertyGetterOnly { get; }
        public string PropertySetterOnly { set { } }
        public int this[int index]
        {
            get { return index; }
            set { index = value; }
        }
        protected int ProtectedProperty { get; set; }
        public static int PropertyStaticPublic { get; set; }
        private static int PropertyStaticPrivate { get; set; }
        internal static int PropertyStaticInternal { get; set; }

        public readonly SampleClass_Public _public;
        public SampleClass_Public _public2;
        private static readonly SampleClass_Public _static;

        public void MethodVoid()
        {
        }

        public static void MethodStaticVoid()
        {
        }
        public void MethodInput(string input1, string input2)
        {
        }
        public void MethodInput<T>(string input1, T input2)
        {
        }

        [return: ReturnAttrib]
        public int MethodIntReturn()
        {
            return 0;
        }
    }
    public class ReturnAttribAttribute : Attribute { }

    public class SampleClass_Public
    {
    }

    public interface SampleInterface
    {
         int Property{ get; set; }
    }

    public class SampleClass_Generic<T>
    {
        T Property { get; set; }
    }
    internal class SampleClass_Internal
    {
    }
    public sealed class SampleClass_PublicSealed
    {
    }
    public abstract class SampleClass_PublicAbstract
    {
        internal class Nested
        {
        }
    }
    public static class SampleClass_PublicStatic
    {
    }
    public struct SampleStruct
    {
    }
    public enum SampleEnum
    {
        V1,
        V2,
    }
    public enum SampleByteEnum : byte
    {
        V1,
        V2,
    }
}

File: src/BinaryDataDecoders.CodeAnalysis.Tests/VisualBasic/SampleClasses.vb
Imports System
Imports System.Collections.Generic
Imports System.Runtime.InteropServices
Imports System.Text

Namespace BinaryDataDecoders.CodeAnalysis.Tests.CSharp
    <Serializable>
    Class SampleClass_NoVisibility
        <Obsolete>
        Public Sub New()
        End Sub

        Public Sub New(
<[In]> ByVal value As Integer)
        End Sub

        Private Shared Sub New()
        End Sub

        Protected Overrides Sub Finalize()
        End Sub

        Public ReadOnly Property PropertyGetterOnlyArrow As String
            Get
                Return Nothing
            End Get
        End Property

        Public ReadOnly Property PropertyGetterOnly As String

        Public WriteOnly Property PropertySetterOnly As String
            Set(ByVal value As String)
            End Set
        End Property

        Default Public Property Item(ByVal index As Integer) As Integer
            Get
                Return index
            End Get
            Set(ByVal value As Integer)
                index = value
            End Set
        End Property

        Protected Property ProtectedProperty As Integer
        Public Shared Property PropertyStaticPublic As Integer
        Private Shared Property PropertyStaticPrivate As Integer
        Friend Shared Property PropertyStaticInternal As Integer
        Public ReadOnly _public As SampleClass_Public
        Public _public2 As SampleClass_Public
        Private Shared ReadOnly _static As SampleClass_Public

        Public Sub MethodVoid()
        End Sub

        Public Shared Sub MethodStaticVoid()
        End Sub

        Public Sub MethodInput(ByVal input1 As String, ByVal input2 As String)
        End Sub

        Public Sub MethodInput(Of T)(ByVal input1 As String, ByVal input2 As T)
        End Sub

        <ReturnAttrib>
        Public Function MethodIntReturn() As Integer
            Return 0
        End Function
    End Class

    Public Class ReturnAttribAttribute
        Inherits Attribute
    End Class

    Public Class SampleClass_Public
    End Class

    Interface SampleInterface
        Private Property [Property] As Integer
    End Interface

    Public Class SampleClass_Generic(Of T)
        Private Property [Property] As T
    End Class

    Friend Class SampleClass_Internal
    End Class

    Public NotInheritable Class SampleClass_PublicSealed
    End Class

    Public MustInherit Class SampleClass_PublicAbstract
        Friend Class Nested
        End Class
    End Class

    Module SampleClass_PublicStatic
    End Module

    Public Structure SampleStruct
    End Structure

    Public Enum SampleEnum
        V1
        V2
    End Enum
End Namespace
